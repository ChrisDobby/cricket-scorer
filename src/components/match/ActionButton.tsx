import * as React from 'react';

const buttonStyle: React.CSSProperties = {
    minWidth: '36px',
    margin: '2px',
};

export interface ActionButtonProps {
    caption: string;
    noBall: boolean;
    action: () => void;
}

export const ActionButton = ({ caption, noBall, action }: ActionButtonProps) => (
    <button
        className={`btn ${noBall ? 'btn-danger' : 'btn-success'}`}
        style={buttonStyle}
        onClick={action}
    >{caption}
    </button>
);
