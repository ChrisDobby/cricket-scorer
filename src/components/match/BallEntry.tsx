import * as React from 'react';
import * as globalStyles from '../styles';
import { Innings, Batter, Bowler, oversDescription, BattingInnings } from '../../domain';

const battingScoreStyle: React.CSSProperties = {
    display: 'inline',
    marginRight: '10px',
};

export interface BallEntryProps {
    innings: Innings;
    batter: Batter;
    bowler: Bowler;
}

export const BallEntry = ({ innings, batter, bowler }: BallEntryProps) => (
    <div style={globalStyles.sectionContainer}>
        <div className="row" style={globalStyles.singleHeadingRow}>
            <h4>{`${bowler.name} to ${batter.name}`}</h4>
        </div>
        <div className="row">
            <div className="col-6">
                <h5>{innings.battingTeam.name}</h5>
            </div>
            <div className="col-6">
                <h5>{`${innings.score}-${innings.wickets}`}</h5>
            </div>
        </div>
        <div className="row">
            <div className="col-6">
                <h5>Overs</h5>
            </div>
            <div className="col-6">
                <h5>{oversDescription(innings.balls)}</h5>
            </div>
        </div>
        {innings.batting.batters.filter(batter => batter.innings && !batter.innings.wicket)
            .map(batter => ({
                name: batter.name,
                innings: batter.innings as BattingInnings,
            }))
            .map(currentBatter => (
                <div className="row">
                    <div className="col-6">
                        <h5>{currentBatter.name}</h5>
                    </div>
                    <div className="col-6">
                        <span>
                            <h5 style={battingScoreStyle}>{currentBatter.innings.runs}</h5>
                            {currentBatter.name === batter.name && <i className="fa fa-circle" />}
                        </span>
                    </div>
                </div>
            ))}
        <div className="row">
            <div className="col-6">
                <h5>{bowler.name}</h5>
            </div>
            <div className="col-6">
                <h5>{`${oversDescription(bowler.balls)}-${bowler.maidenOvers}-${bowler.runs}-${bowler.wickets}`}</h5>
            </div>
        </div>
    </div>
);
