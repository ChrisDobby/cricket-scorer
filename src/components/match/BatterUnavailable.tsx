import * as React from 'react';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogActions from '@material-ui/core/DialogActions';
import Button from '@material-ui/core/Button';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import { UnavailableReason, Innings } from '../../domain';

interface BatterUnavailableProps {
    reason: UnavailableReason;
    innings: Innings;
    battingPlayers: string[];
    update: (playerIndex: number, reason: UnavailableReason) => void;
    cancel: () => void;
}

export default (props: BatterUnavailableProps) => {
    const [playerIndex, setPlayerIndex] = React.useState(undefined as number | undefined);

    const batters = props.innings.batting.batters.filter(
        batter =>
            (props.reason === UnavailableReason.Absent && !batter.innings) ||
            (props.reason === UnavailableReason.Retired && batter.innings && !batter.innings.wicket),
    );
    const description = props.reason === UnavailableReason.Retired ? 'retired' : 'absent';

    return (
        <div>
            <Dialog open={true} aria-labelledby="update-innings-overs-title">
                <DialogTitle id="update-innings-overs-title">{`Mark one of the batting side as ${description}`}</DialogTitle>
                <DialogContent>
                    <DialogContentText>{`Select the ${description} player`}</DialogContentText>
                </DialogContent>
                <DialogContent>
                    {batters.length > 0 && (
                        <Select value={playerIndex} onChange={ev => setPlayerIndex(Number(ev.target.value))} fullWidth>
                            {batters.map(batter => (
                                <MenuItem key={batter.playerIndex} value={batter.playerIndex}>
                                    {props.battingPlayers[batter.playerIndex]}
                                </MenuItem>
                            ))}
                        </Select>
                    )}
                    {batters.length === 0 && (
                        <DialogContentText>{`No players can be marked as ${description}`}</DialogContentText>
                    )}
                </DialogContent>
                {batters.length > 0 && (
                    <DialogActions>
                        <Button
                            onClick={() => props.update(playerIndex as number, props.reason)}
                            color="primary"
                            autoFocus
                            disabled={typeof playerIndex === 'undefined'}
                        >
                            OK
                        </Button>
                        <Button onClick={props.cancel} color="primary">
                            Cancel
                        </Button>
                    </DialogActions>
                )}
                {batters.length === 0 && (
                    <DialogActions>
                        <Button onClick={props.cancel} color="primary" autoFocus>
                            OK
                        </Button>
                    </DialogActions>
                )}
            </Dialog>
        </div>
    );
};
