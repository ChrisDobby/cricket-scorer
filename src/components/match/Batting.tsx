import * as React from 'react';
import { Extras } from './Extras';
import { Batting as InningsBatting, BattingInnings, howOutDescription } from '../../domain';
import * as styles from './styles';

const battingContainer: React.CSSProperties = {
    width: '100%',
    padding: '20px',
    marginBottom: '20px',
};

const howOut = (innings?: BattingInnings): string => {
    if (!innings) { return ''; }

    return howOutDescription(innings.wicket);
};

interface InningsItemProps { innings?: BattingInnings; }
const Howout = (props: InningsItemProps) => (
    <div className="col-md-4">{howOut(props.innings)}</div>
);

interface CellItemProps {
    property: string;
    style: React.CSSProperties;
}
const CellItem = ({ property, style }: CellItemProps) => (
    <div className="col-md-1" style={style}>{property}</div>
);

export interface BattingProps { batting: InningsBatting; }

export const Batting = ({ batting }: BattingProps) => (
    <div className="col-xl-8 col-lg-12">
        <div style={battingContainer}>
            <div className="row" style={styles.headingRow}>
                <div className="col-md-7"><h6>Batsman</h6></div>
                <div className="col-md-1" style={styles.numberCell}><h6>Runs</h6></div>
                <div className="col-md-1" style={styles.numberCell}><h6>Balls</h6></div>
                <div className="col-md-1" style={styles.numberCell}><h6>Mins</h6></div>
                <div className="col-md-1" style={styles.numberCell}><h6>4s</h6></div>
                <div className="col-md-1" style={styles.numberCell}><h6>6s</h6></div>
            </div>    
            {batting.batters.map(batter =>
                <div className="row" style={styles.itemRow} key={batter.position}>
                    <div className="col-md-3">{batter.name}</div>
                    <Howout innings={batter.innings} />
                    <CellItem property={batter.innings ? batter.innings.runs.toString() : ''} style={styles.runsCell} />
                    <CellItem
                        property={batter.innings ? batter.innings.ballsFaced.toString() : ''}
                        style={styles.numberCell}
                    />
                    <CellItem property={batter.innings ? '0' : ''} style={styles.numberCell} />
                    <CellItem
                        property={batter.innings ? batter.innings.fours.toString() : ''}
                        style={styles.numberCell}
                    />
                    <CellItem
                        property={batter.innings ? batter.innings.sixes.toString() : ''}
                        style={styles.numberCell}
                    />
                </div>)}
            <Extras extras={batting.extras} />
        </div>
    </div>
);
