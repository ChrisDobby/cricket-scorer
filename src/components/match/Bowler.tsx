import * as React from 'react';
import * as styles from './styles';
import * as globalStyles from '../styles';

const indicatorStyle: React.CSSProperties = {
    ...globalStyles.primaryColour,
    float: 'left',
    width: '30px',
    textAlign: 'center',
};

const nonSelectableStyle: React.CSSProperties = {
    color: 'rgb(233, 236, 239)',
};

interface SelectionIndicatorProps {
    playerIndex: number;
    selected: boolean;
}

const SelectionIndicator = ({ playerIndex, selected }: SelectionIndicatorProps) => {
    return (
        <div style={indicatorStyle}>
            {selected && <i className="fa fa-circle" />}
        </div>
    );
};

export interface BowlerProps {
    index: number;
    selected: boolean;
    name: string;
    allowed: boolean;
    selectBowler: (bowler: number) => void;
}

export const Bowler = ({ index, selected, name, allowed, selectBowler }: BowlerProps) => (
    <div
        className="row bowler-row"
        style={allowed ? styles.selectablePlayerStyle : nonSelectableStyle}
        onClick={() => { if (allowed) { selectBowler(index); } }}
    >
        <SelectionIndicator
            playerIndex={index}
            selected={selected}
        />
        <h6>{name}</h6>
    </div>
);
