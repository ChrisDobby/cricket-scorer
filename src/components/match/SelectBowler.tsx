import * as React from 'react';
import { Team } from '../../domain';
import * as styles from './styles';
import * as globalStyles from '../styles';
import { SaveButton } from './SaveButton';

const indicatorStyle: React.CSSProperties = {
    float: 'left',
    width: '30px',
    textAlign: 'center',
};

interface SelectionIndicatorProps {
    playerIndex: number;
    selectedIndex: number;
}

const SelectionIndicator = ({ playerIndex, selectedIndex }: SelectionIndicatorProps) => {
    return (
        <div style={indicatorStyle}>
            {playerIndex === selectedIndex &&
                <i className="fa fa-circle" />}
        </div>
    );
};

export interface SelectBowlerProps {
    bowlingTeam: Team;
    selectBowler: (b: number) => void;
}

export class SelectBowler extends React.Component<SelectBowlerProps, {}> {
    state = { selectedPlayerIndex: -1 };

    save = () => {
        if (this.state.selectedPlayerIndex >= 0) {
            this.props.selectBowler(this.state.selectedPlayerIndex);
        }
    }

    selectBowler = (playerIndex: number) => {
        this.setState({
            selectedPlayerIndex: playerIndex,
        });
    }

    get canSave() {
        return this.state.selectedPlayerIndex >= 0;
    }

    render() {
        return (
            <div>
                <div className="row">
                    <div className="d-none d-md-block d-lg-block col-2 col-lg-3" />
                    <div className="col-12 col-md-8 col-lg-6">
                        <div style={globalStyles.sectionContainer}>
                            <div className="row" style={globalStyles.headingRow}>
                                <h4>Select bowler</h4>
                            </div>
                            {this.props.bowlingTeam.players.map((player, index) => (
                                <div
                                    key={player}
                                    className="row bowler-row"
                                    style={styles.selectablePlayerStyle}
                                    onClick={() => this.selectBowler(index)}
                                >
                                    <SelectionIndicator
                                        playerIndex={index}
                                        selectedIndex={this.state.selectedPlayerIndex}
                                    />
                                    <h6>{player}</h6>
                                </div>
                            ))}
                        </div>
                    </div>
                </div>
                <SaveButton enabled={this.canSave} save={this.save} />
            </div>
        );
    }
}

