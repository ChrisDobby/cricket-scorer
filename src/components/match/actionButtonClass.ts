import { DeliveryOutcome } from '../../domain';

const actionButtonClass = (invalidClassName: string) => (deliveryOutcome: DeliveryOutcome) =>
    deliveryOutcome === DeliveryOutcome.Noball || deliveryOutcome === DeliveryOutcome.Wide
        ? invalidClassName
        : undefined;

export default actionButtonClass('btn-danger');
