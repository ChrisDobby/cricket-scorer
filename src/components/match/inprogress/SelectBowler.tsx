import * as React from 'react';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Grid from '@material-ui/core/Grid';
import { Team } from '../../../domain';
import EditForm from '../EditForm';

interface SelectBowlerProps {
    bowlingTeam: Team;
    disallowedPlayers: number[];
    initiallySelected?: number;
    selectBowler: (b: number) => void;
}

export default (props: SelectBowlerProps) => {
    const [selectedPlayerIndex, setSelectedPlayerIndex] = React.useState(
        typeof props.initiallySelected === 'undefined' ? -1 : props.initiallySelected,
    );

    const canSave = () => selectedPlayerIndex >= 0;
    const save = () => {
        if (selectedPlayerIndex >= 0) {
            props.selectBowler(selectedPlayerIndex);
        }
    };

    return (
        <Grid container>
            <Grid item sm={1} md={2} />
            <Grid item xs={12} sm={10} md={8}>
                <EditForm heading="Select bowler" save={save} canSave={canSave}>
                    <List>
                        {props.bowlingTeam.players.map((player, index) => (
                            <ListItem
                                disabled={props.disallowedPlayers.indexOf(index) >= 0}
                                selected={index === selectedPlayerIndex}
                                key={index}
                                role={undefined}
                                dense
                                button
                                color="primary"
                                onClick={() => setSelectedPlayerIndex(index)}
                            >
                                <ListItemText primary={player} />
                            </ListItem>
                        ))}
                    </List>
                </EditForm>
            </Grid>
        </Grid>
    );
};
