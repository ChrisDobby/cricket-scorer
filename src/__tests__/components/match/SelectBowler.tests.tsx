import * as React from 'react';
import { shallow } from 'enzyme';
import * as ReactTestRenderer from 'react-test-renderer';
import { SelectBowler } from '../../../components/match/SelectBowler';
import { match } from '../../testData/testMatch';

describe('SelectBowler', () => {
    const players = [
        'Player 1',
        'Player 2',
        'Player 3',
        'Player 4',
        'Player 5',
        'Player 6',
        'Player 7',
        'Player 8',
        'Player 9',
        'Player 10',
        'Player 11',
    ];

    const team = {
        ...match.homeTeam,
        players,
    };

    const select = jest.fn();

    beforeEach(() => jest.resetAllMocks());

    const selectBowler = shallow(<SelectBowler bowlingTeam={team} selectBowler={select} disallowedPlayers={[]} />);

    it('should update state when selectBowler called', () => {
        const instance = selectBowler.instance() as SelectBowler;
        instance.selectBowler(0);

        expect(selectBowler.state().selectedPlayerIndex).toBe(0);
    });

    it('should be able to save when a player is selected', () => {
        selectBowler.setState({
            selectedPlayerIndex: 0,
        });

        const instance = selectBowler.instance() as SelectBowler;
        expect(instance.canSave).toBeTruthy();
    });

    it('should not be able to save when a player is not selected', () => {
        selectBowler.setState({
            selectedPlayerIndex: -1,
        });

        const instance = selectBowler.instance() as SelectBowler;
        expect(instance.canSave).toBeFalsy();
    });

    it('should call select func when save called if a player is selected', () => {
        selectBowler.setState({
            selectedPlayerIndex: 0,
        });

        const instance = selectBowler.instance() as SelectBowler;
        instance.save();

        expect(select).toHaveBeenCalledWith(0);
    });

    it('should not call select func when save called if no player selected', () => {
        selectBowler.setState({
            selectedPlayerIndex: -1,
        });

        const instance = selectBowler.instance() as SelectBowler;
        instance.save();

        expect(select).toHaveBeenCalledTimes(0);
    });

    it('should render correctly when no player selected', () => {
        const selectBowler = ReactTestRenderer
            .create(<SelectBowler bowlingTeam={team} selectBowler={select} disallowedPlayers={[]} />);

        expect(selectBowler.toJSON()).toMatchSnapshot();
    });

    it('should render correctly when a player is selected', () => {
        const selectBowler = ReactTestRenderer
            .create(<SelectBowler bowlingTeam={team} selectBowler={select} disallowedPlayers={[]} />);

        selectBowler.root.instance.setState({
            selectedPlayerIndex: 6,
        });

        expect(selectBowler.toJSON()).toMatchSnapshot();
    });

    it('should render correctly when a player is disallowed', () => {
        const selectBowler = ReactTestRenderer
            .create(<SelectBowler bowlingTeam={team} selectBowler={select} disallowedPlayers={[0]} />);

        expect(selectBowler.toJSON()).toMatchSnapshot();
    });
});
