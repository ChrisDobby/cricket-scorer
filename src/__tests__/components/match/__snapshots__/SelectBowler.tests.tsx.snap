// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBowler should render correctly when a player is selected 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="d-none d-md-block d-lg-block col-2 col-lg-3"
    />
    <div
      className="col-12 col-md-8 col-lg-6"
    >
      <div
        style={
          Object {
            "marginBottom": "10px",
            "padding": "20px",
            "width": "100%",
          }
        }
      >
        <div
          className="row"
          style={
            Object {
              "backgroundColor": "rgb(233, 236, 239)",
              "paddingLeft": "15px",
              "paddingTop": "4px",
            }
          }
        >
          <h4>
            Select bowler
          </h4>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 1
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 2
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 3
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 4
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 5
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 6
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          >
            <i
              className="fa fa-circle"
            />
          </div>
          <h6>
            Player 7
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 8
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 9
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 10
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 11
          </h6>
        </div>
      </div>
    </div>
  </div>
  <button
    className="btn btn-success"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "borderRadius": "25px",
        "bottom": "10px",
        "boxShadow": "0 4px 6px 0 rgba(0,0,0,.14), 0 4px 5px rgba(0,0,0,-1)",
        "fontSize": "18px",
        "height": "50px",
        "lineHeight": "1.33",
        "padding": "10px 16px",
        "position": "fixed",
        "right": "10px",
        "width": "50px",
        "zIndex": 1049,
      }
    }
    type="button"
  >
    <i
      className="fa fa-check"
    />
  </button>
</div>
`;

exports[`SelectBowler should render correctly when no player selected 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="d-none d-md-block d-lg-block col-2 col-lg-3"
    />
    <div
      className="col-12 col-md-8 col-lg-6"
    >
      <div
        style={
          Object {
            "marginBottom": "10px",
            "padding": "20px",
            "width": "100%",
          }
        }
      >
        <div
          className="row"
          style={
            Object {
              "backgroundColor": "rgb(233, 236, 239)",
              "paddingLeft": "15px",
              "paddingTop": "4px",
            }
          }
        >
          <h4>
            Select bowler
          </h4>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 1
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 2
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 3
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 4
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 5
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 6
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 7
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 8
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 9
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 10
          </h6>
        </div>
        <div
          className="row bowler-row"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <div
            style={
              Object {
                "float": "left",
                "textAlign": "center",
                "width": "30px",
              }
            }
          />
          <h6>
            Player 11
          </h6>
        </div>
      </div>
    </div>
  </div>
  <button
    className="btn btn-success"
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "borderRadius": "25px",
        "bottom": "10px",
        "boxShadow": "0 4px 6px 0 rgba(0,0,0,.14), 0 4px 5px rgba(0,0,0,-1)",
        "fontSize": "18px",
        "height": "50px",
        "lineHeight": "1.33",
        "padding": "10px 16px",
        "position": "fixed",
        "right": "10px",
        "width": "50px",
        "zIndex": 1049,
      }
    }
    type="button"
  >
    <i
      className="fa fa-check"
    />
  </button>
</div>
`;
